openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/CpuArchitecture.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * The copyright of this file belongs to Feedzai. The file cannot be
+ * reproduced in whole or in part, stored in a retrieval system,
+ * transmitted in any form, or by any means electronic, mechanical,
+ * photocopying, or otherwise, without the prior permission of the owner.
+ *
+ * © 2023 Feedzai, Strictly Confidential
+ */
+package com.feedzai.openml.provider.lightgbm;
+
+/**
+ * Enum that represents the cpu architecture where code is running and consequent lgbm native libs locations.
+ */
+public enum CpuArchitecture {
+    AARCH64("arm64"),
+    AMD64("amd64");
```

Comment:
I'm a bit worried about this: https://github.com/openjdk/jdk/blob/master/make/autoconf/platform.m4#L481C2-L481C2

it maybe not be critical it now use case but I'm afraid we are not capturing all the possibilities on this and then we need to deal with it getting all the context on this. At least some documentation why are we only considering these 2 options.

Reply:
We just support lightgbm compiled for amd64 and aarch64. From the tests I've done on the two architectures, I saw that the JVM set the os.arch to either amd64 on x86_64 or aarch64 on aarch64 cpus... I've always used zulu on the tests with arm64 and openjdk on x86_64.

Reply:
Please make sure that is  very well captured on the documentation.

Reply:
It's in the README